### DISCUSSION TITLE
Feature request for "For You(beta)"

### DISCUSSION THREAD

**Discussion by WingTechCorner** [community/community#13354] [Other Feature Feedback, Questions, & Ideas]
# Feature Request for "For You(beta)"

Hello Github, I saw the news about the "For You(beta)" being added and have been taking a look at it.

I have a few features I would like to request and they are listed below

1. Please make the "suggestions" in the feed less "Facebook"-like. I don't need or want what is essentially data-mined and algorithm inferred suggestions for things/projects/people to follow. I can make my own decisions and have done so, to ensure I have a good signal to noise ratio.
2. Please make the block/module *OPT-IN* and not *OPT-OUT* 
3. Dovetailing on the first item above, please also refrain from mimicking behaviors from other social networks: Twitter, TikTok, WeChat, Tinder, Grindr, etc.  That is all noise that is fine on their respective platforms, but is essentially 100% noise for me here. There is NULL value add.

# Rationale For Feature Request for "For You(beta)"

1. Github is a place that has been great for focusing almost exclusively on *coding*. If I wanted a social feed, I know where to go to get it and I have significantly better customization features there. I do *NOT* need to go through the teething pains of Github figuring out how to make its own social feed palatable.
2. Noise. Funny thing, people automate against Github. Don't need noise. Don't need links and references to projects I'm not monitoring or being spammed by changes/events that I have already determined how much/little I want to hear from. Adding the functionality under a tab isn't really any better. I'm already annoyed by differences in the Github interface between different platforms/etc, please don't add another annoyance.
3. Another avenue for phishing/scams to be distributed. You _KNOW_ people whose job it is to find and leverage such algorithmically controlled feeds to send unwanted information to people. Please do not provide another outlet for this noise/sewage.

# Questions about "For You(beta)"

I'm curious as to where the idea for this functionality came from. What was the impetus for the creation of a feed of this nature? What was the outlined value add to the user base? What was the consumer profile that this feature was designed against/for? I'm very curious to know, what kind of envisioned software engineer/GitHub user would have found such a feature useful or would have actually asked for such a feature to be auto-opted in for everyone as a beta release?

# Please consider a closed beta in the future (and perhaps move this feature into a closed beta)

I don't like being a beta-user when I haven't been given the choice. By changing the interface so that it is prominently displayed in the central/above-fold location, puts the feature central in one's vision. Ie, Github is essentially telling me: Hey, hey, I have this cool feature I made, please stop what you are doing and mess with this! Really, ignore the alerts and be distracted from your actual work/code/pipeline related feeds and read some of this interesting stuff that we had software pick out for you based on a set of rules we aren't explaining. But we want to mix in the updates from other channels all into one channel so that it becomes even more difficult and annoying.

No, please do this in a closed beta next time. And when it makes it out of beta, then make it an opt-in module.

# If you are thinking of creating a monetization channel....

If this is a prelude to a monetization stream, please don't. Full stop. 

I expect Github to be stable and reliable. (coincidentally, the last couple of days, it has not been) Adding in any kind of ad means adding in additional infrastructure and/or dependencies to the platform/interface that will either introduce points or sources of failure... or it will result in personal data being exposed to third parties.

Not that people should be keeping any kind of tokens or passwords in their repo, but I don't need or want my code base to be skimmed like that. If I did, I would have done my code publishing on Facebook, Google, Twitter, etc.

# Closing Thought

To be honest, I never thought this would be a point of concern with Github. I'm saddened and disappointed by the fact that this has arisen. The reason being that I have to judge a platform by its actions and by any changes it makes. As those changes and actions predict where the platform is intending to go.  And the creation of an unwanted social feed as a public opt-out beta... has all the signs of a desire to monetize traffic going through Github's various pages.  Starts on a feed, but eventually will be on each and every page, with the promise of no-storing of private details as the backend performs scans with potential offload to third party services. I imagine this showing up on both public and private repos. 

And if that is the direction even considered, please don't. Just don't. Stop and consider what the use case of GitHub is, who the users are, why they are here. I personally don't see any value add. Rather, I see such an added function to be subtractive from the perceived value of GitHub.

**Comment by jfullstackdev** [comment]
there is a new update on the For You feed, you can now filter it, and for those who don't want it can simply disable it by unchecking all filters

![image](https://user-images.githubusercontent.com/47092464/182012117-0b19be9d-e849-462e-909e-d20d28915c1d.png)
